examples/connext_dds/xml_application_env/c++11/xml_application_env.cpp

```diff
@@ -0,0 +1,150 @@
+                       .time_since_epoch()
+                       .count();
+    sample.value<int64_t>("time", now);
+
+    char buf[4096];
+
+    // Loop over the env pointer, extract the key and value from the enviroment
+    // variable, set the values in the sample, and write with the writer.
+    while (*env) {
+        try {
+            strncpy(buf, *env, 4095);
+            buf[4095] = '\0';  // prevent buffer overflow
+            char *key = strtok(buf, "=");
+            char *value = strtok(NULL, "=");
+            sample.value<std::string>("key", key);
+            sample.value<std::string>("value", value);
+            env++;
+
+            // Now we can write out the populated sample of a single environment
+            // variable.
+            writer.write(sample);
+        } catch (std::exception ex) {
+            std::cout << "Exception caught publishing sample: " << ex.what()
+                      << std::endl;
+        }
+    }
+}
+
+
+int main(int argc, char *argv[], char *envp[])
+{
+    // Create the QosPovider from the specified xml file.
+    // The application.xml filename is used in this example instead of the
+    // default.
+    QosProvider qos_provider(QOS_URL);
+
+    // Create the participant as defined in the xml file.  This instantiats the
+    // participant registeres the types, and creates the child publisher,
+    // subscriber, reader and writer.
+    DomainParticipant participant =
+            qos_provider->create_participant_from_config(PARTICIPANT_NAME);
+
+    // Lookup the specific KeyValuePair type as defined in the xml file.
+    // This will be needed to create samples of the correct type when
+    // publishing.
+    const DynamicType &myType = qos_provider->type(KVP_TYPE_NAME);
+
+    // Find the DataWriter defined in the xml by using the participant and the
+    // publisher::writer pair as the datawriter name.
+    DataWriter<DynamicData> writer =
+            rti::pub::find_datawriter_by_name<DataWriter<DynamicData>>(
+                    participant,
+                    WRITER_NAME);
+
+    // Find the DataReader defined in the xml by using the participant and the
+    // subscriber::reader pair as the datareader name.
+    DataReader<DynamicData> reader =
+            rti::sub::find_datareader_by_name<DataReader<DynamicData>>(
+                    participant,
+                    READER_NAME);
+
+    // Associate the listener to the data reader.
+    ListenerBinder<DataReader<DynamicData>> datareader_listener =
+            rti::core::bind_and_manage_listener(
+                    reader,
+                    new MyDataReaderListener,
+                    dds::core::status::StatusMask::data_available());
+
+    // Ignore our own publications so they do not trigger our data listener.
+    dds::domain::ignore(participant, participant.instance_handle());
+
+    // Application id to uniquely identify data from a particular application
+    int id = std::rand();
```

Comment:
What's the purpose of this ID on the other side? Often we try to decouple the information we publish from the actual source, so I wanted to understand what's its purpose and if there's something already in DDS we could reuse.